### PR TITLE
Culling portals

### PR DESCRIPTION
**Pull request by MitjaBezensek**
Describe what your pull request does. If appropriate, add GIFs or images showing the before and after.

### Change Type

- [ ] `sdk` — Changes the tldraw SDK
- [ ] `dotcom` — Changes the tldraw.com web app
- [ ] `docs` — Changes to the documentation, examples, or templates.
- [ ] `vs code` — Changes to the vscode plugin
- [ ] `internal` — Does not affect user-facing stuff

- [ ] `bugfix` — Bug fix
- [ ] `feature` — New feature
- [ ] `improvement` — Improving existing features
- [ ] `chore` — Updating dependencies, other boring stuff
- [ ] `galaxy brain` — Architectural changes
- [ ] `tests` — Changes to any test code
- [ ] `tools` — Changes to infrastructure, CI, internal scripts, debugging tools, etc.
- [ ] `dunno` — I don't know

### Test Plan

1. Add a step-by-step description of how to test your PR here.
2.

- [ ] Unit Tests
- [ ] End to end tests

### Release Notes

- Add a brief release note for your PR here.

### CODE REVIEW

File: packages/editor/src/lib/components/default-components/DefaultCanvas.tsx [hunk 1]
```diff
@@ -364,13 +368,59 @@ function ShapesToDisplay() {
...
+				wrapperDiv.current?.appendChild(shapePortalInfo.container)
+				portalInfo.set(shape.id, { container: shapePortalInfo.container, addedToDom: true })
+			}
+		})
+	}
+
+	useEffect(() => {
+		function cleanContainerDivs() {
+			if (!wrapperDiv.current) return
+			const divIdsToRemove = new Set(portalInfo.keys())
```
Comment (MitjaBezensek) [inline packages/editor/src/lib/components/default-components/DefaultCanvas.tsx:396]: Could also get the ids from dom, would be more accurate 🤷‍♂️

**Comment by mimecuvalo** [comment]
out of curiosity: how was the perf on this one?

**Comment by MitjaBezensek** [comment]
> out of curiosity: how was the perf on this one?

Quite good. There's a comparison between the different approaches [here](https://github.com/tldraw/tldraw/pull/3291).

**Comment by MitjaBezensek** [comment]
Closing in favor of display none.

**Pull request closed without merging**